include/zim/writer/creator.h

```diff
@@ -137,6 +137,23 @@ namespace zim
          */
         void addMetadata(const std::string& name, std::unique_ptr<ContentProvider> provider, const std::string& mimetype);
 
+        /**
+         * Add illustration to the archive.
+         *
+         * @param width the width of the illustration.
+         * @param height the height of the illustration.
+         * @param content the content of the illustration (must be a png content)
+         */
+        void addIllustration(unsigned int width, unsigned int height, const std::string& content);
```

Comment:
If content is required to be a PNG image, asking for its dimensions in the interface signature is redundant. A safer interface is to validate the content independently and extract its size automatically. Of course doing this will add to libzim a new dependency (libpng), however it will remove the burden from future clients of this interface.